Spoony-iOS/Spoony-iOS/Source/Feature/Report/Report.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  Report.swift
+//  Spoony-iOS
+//
+//  Created by 최주리 on 1/16/25.
+//
+
+import SwiftUI
+
+enum ReportType: String, CaseIterable {
+    case commercial = "영리 목적/ 홍보성 후기"
+    case abusiveLanguage = "욕설/인신공격"
+    case illegalInformation = "불법정보"
+    case personalInformationExposure = "개인정보노출"
+    case spam = "도배"
+    case others = "기타"
+}
+
+struct Report: View {
+    @State private var selectedReport: ReportType = .commercial
+    @State private var text: String = ""
+    @State private var errorState: TextEditorErrorState = .noError
+    @State private var isDisabled: Bool = true
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            //TODO: 변경 예정
+            CustomNavigationBar(style: .primary, title: "신고하기") {
+                
+            }
+            Divider()
+            
+            reportTitle
+            reportDescription
+            
+            SpoonyButton(
+                style: .secondary,
+                size: .xlarge,
+                title: "신고하기",
+                disabled: $isDisabled
+            ) {
+                
+            }
+            .padding(.top, errorState == .noError ? 12 : 20)
+            .padding(.bottom, 20)
+        }
+        .frame(maxHeight: .infinity)
+        .onTapGesture {
+            if errorState == .maximumInputError(style: .report) {
+                errorState = .noError
+            }
+        }
+        .onChange(of: errorState) {
+            if errorState == .minimumInputError(style: .report) {
+                isDisabled = true
+            } else {
+                isDisabled = false
+            }
+            
+            if errorState == .maximumInputError(style: .report) {
+                isDisabled = true
+            }
+        }
```

Comment:
중복 코드 줄여도 좋을 것 같습니다.

```suggestion
        .onChange(of: errorState) {
            if errorState == .minimumInputError(style: .report) || errorState == .maximumInputError(style: .report) {
                isDisabled = true
            } else {
                isDisabled = false
            }
        }
```